File: fluentlenium-assertj/src/test/java/io/fluentlenium/assertj/integration/page/pages/IndexPage.java
package io.fluentlenium.assertj.integration.page.pages;

import io.fluentlenium.core.FluentPage;
import io.fluentlenium.core.domain.FluentList;
import io.fluentlenium.core.domain.FluentWebElement;
import io.fluentlenium.core.FluentPage;
import io.fluentlenium.core.domain.FluentList;
import io.fluentlenium.core.domain.FluentWebElement;
import org.openqa.selenium.support.FindAll;
import org.openqa.selenium.support.FindBy;

import static io.fluentlenium.assertj.FluentLeniumAssertions.assertThat;

@FindAll({@FindBy(id = "oneline"), @FindBy(className = "small")})
public class IndexPage extends FluentPage {

    @FindBy(id = "oneline")
    private FluentWebElement fluentWebElement;

    @FindBy(className = "small")
    private FluentList<FluentWebElement> fluentList;

    public void verifyElement() {
        assertThat(this).hasElement(fluentWebElement);
    }

    public void verifyElements() {
        assertThat(this).hasElements(fluentList);
    }

    public void verifyElementDisplayed() {
        assertThat(this).hasElementDisplayed(fluentWebElement);
    }
}

File: fluentlenium-kotest-assertions/src/test/kotlin/io/fluentlenium/kotest/matchers/config/pages/IndexPage.kt
package io.fluentlenium.kotest.matchers.config.pages

import io.kotest.matchers.should
import io.fluentlenium.core.FluentPage
import io.fluentlenium.core.domain.FluentList
import io.fluentlenium.core.domain.FluentWebElement
import org.fluentlenium.kotest.matchers.page.haveElement
import org.fluentlenium.kotest.matchers.page.haveElementDisplayed
import org.fluentlenium.kotest.matchers.page.haveElements
import org.openqa.selenium.support.FindAll
import org.openqa.selenium.support.FindBy

@FindAll(FindBy(id = "oneline"), FindBy(className = "small"))
class IndexPage : _root_ide_package_.io.fluentlenium.core.FluentPage() {
    @FindBy(id = "oneline")
    lateinit var fluentWebElement: _root_ide_package_.io.fluentlenium.core.domain.FluentWebElement

    @FindBy(className = "small")
    lateinit var fluentList: _root_ide_package_.io.fluentlenium.core.domain.FluentList<_root_ide_package_.io.fluentlenium.core.domain.FluentWebElement>

    fun verifyElement() {
        this should haveElement(fluentWebElement)
    }

    fun verifyElements() {
        this should haveElements(fluentList)
    }

    fun verifyElementDisplayed() {
        this should haveElementDisplayed(fluentWebElement)
    }
}
